Application hooks

rodw1995 opened the pull request:
### Linked issue
#2 Application Hooks

### Additional context
Not sure what to do with the **package.json**, **CHANGELOG.md** and **README.md** files. Also not yet added to the `use-expo` package. If you can explain what to do to with that files and package, I'm happy to make the required changes.

byCedric commented:
Awesome! ❤️ Thanks for this, I can help with the documentation and packaging. I'll try to make some time available this weekend to get this resolved for you. For now, we can ignore the code climate part that's only a loose sanity check.

rodw1995 commented:
Geen probleem 😉 

- Yes, I saw Lerna and did some reading about it and already thought it would be auto-generating some things. (Good to know that something like Lerna exists btw 😄)

- I committed a readme file and added the application hooks in there. Also created a documentation file per hook like it is done in the other packages.

- The only dev dependency that I have added is `expo-application`.

Yeah, I agree. If more hooks are added it's not a great idea for anyone I think to use the `use-expo` package. So I don't think that should be encouraged.

No, not really weird stuff. Just like I said I didn't know what to do with `use-expo` and the other files.

byCedric commented:
Awesome! Thanks again, I'll add some minor changes to the docblock to add some additional info there. Useful for other developers to just hover over the methods and see what they do 😄 After that I'll release it asap!
